Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionImporter.cs

```diff
@@ -256,6 +256,74 @@ public static void CreateInputAsset()
             ProjectWindowUtil.CreateAssetWithContent("New Controls." + InputActionAsset.Extension,
                 InputActionAsset.kDefaultAssetLayoutJson, InputActionAssetIconLoader.LoadAssetIcon());
         }
+
+        // When an action asset is renamed, copied, or moved in the Editor, the "Name" field in the JSON will
+        // hold the old name and won't match what's in memory (asset looks "dirty"). To work around this, we
+        // must flush the updated JSON to the file whenever this operation occurs.
+        // https://jira.unity3d.com/browse/ISXB-749
+        private class InputActionAssetPostprocessor : AssetPostprocessor
+        {
+            private static List<string> assetFilesNeedingRefresh;
+
+            private void OnPreprocessAsset()
+            {
+                var importer = assetImporter as InputActionImporter;
+                if (importer == null)
+                    return;
+
+                var newName = Path.GetFileNameWithoutExtension(assetPath);
+                var newAsset = ScriptableObject.CreateInstance<InputActionAsset>();
+                var newFileContents = File.ReadAllText(assetPath);
+
+                if (!string.IsNullOrEmpty(newFileContents))
+                {
+                    newAsset.LoadFromJson(newFileContents);
+
+                    // If the serialized Name doesn't match the actual filename this asset file for refresh.
+                    // NOTE: We can't change the file while Asset Importing is in progress.
+                    if (newAsset.name != newName)
+                    {
+                        if (assetFilesNeedingRefresh == null)
+                            assetFilesNeedingRefresh = new List<string>();
+
+                        assetFilesNeedingRefresh.Add(assetPath);
+                    }
+                }
+            }
+
+            private static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths, bool didDomainReload)
+            {
+                if (assetFilesNeedingRefresh == null)
+                    return;
+
+                try
+                {
+                    foreach (var assetPath in assetFilesNeedingRefresh)
+                    {
+                        var newAsset = ScriptableObject.CreateInstance<InputActionAsset>();
+
+                        try
+                        {
+                            var fileContents = File.ReadAllText(assetPath);
+
+                            newAsset.LoadFromJson(fileContents);
+                            newAsset.name = Path.GetFileNameWithoutExtension(assetPath);
```

Comment:
II do not understand why we need to do this at all. This is always unconditionally assigned to InputActionAsset when being imported? The name present in the JSON is, as far as I can tell completely irrelevant or am I missing something?

Reply:
In the importer the JSON name is completely discarded. I guess it only is relevant when loading JSON files manually and parsing them into InputActionAssets if some kind of identification is needed? Not sure where this caused a problem?

Reply:
I see now, it basically generates a diff just because the name is different since the name is adjusted when converted back into JSON..... I guess this fix makes sense then.

Reply:
Wouldn't be cheaper and make more sense to use this kind of processor? https://docs.unity3d.com/ScriptReference/AssetModificationProcessor.OnWillMoveAsset.html

Reply:
As an example see its usage in legacy editor. https://github.com/Unity-Technologies/InputSystem/blob/develop/Packages/com.unity.inputsystem/InputSystem/Editor/AssetEditor/InputActionEditorWindow.cs#L896